OpenNefia.Content/UI/Element/Containers/UiContainer.cs

```diff
@@ -0,0 +1,142 @@
+﻿using OpenNefia.Core;
+using OpenNefia.Core.Maths;
+using OpenNefia.Core.UI.Element;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OpenNefia.Content.UI.Element.Containers
+{
+    public enum LayoutType
+    {
+        None,
+        Element,
+        Spacer,
+        XOffset,
+        YOffset,
+        FullUnset,
+        XUnset,
+        YUnset,
+        YMin,
+        XMin
+    }
+    public class UiContainerEntry
+    {
+        public UiElement? Element { get; set; }
+        public LayoutType Type { get; set;}
+        public int Offset { get; set; }
+
+        public UiContainerEntry(UiElement uiElement)
+        {
+            Element = uiElement;
+            Type = LayoutType.Element;
+        }
+
+        public UiContainerEntry(LayoutType type, int offset)
+        {
+            Type = type;
+            Offset = offset;
+        }
+
+        public static implicit operator UiContainerEntry(UiElement element)
+            => new UiContainerEntry(element);
+    }
+
+    public abstract class UiContainer : UiElement, ILocalizable
+    {
+        protected readonly List<UiContainerEntry> UiElements = new List<UiContainerEntry>();
```

Comment:
Nitpick: Maybe `Entries` since the list doesn't actually hold `UiElements`.

Reply:
yup, were UiElements originally, gonna change it.

Reply:
renamed in 76f6762